File: SiriKit/SiriKitObjCSample/SiriKat/ViewController.m
//
//  ViewController.m
//  SiriKat
//
//  Created by Dobrinka Tabakova on 12/5/16.
//  Copyright © 2016 Dobrinka Tabakova. All rights reserved.
//

#import "ViewController.h"
#import "PhotosProvider.h"
#import "DetailViewController.h"


@interface ViewController () <UITableViewDataSource, UITableViewDelegate>

@property (nonatomic, strong) PhotosProvider *provider;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.provider = [PhotosProvider sharedInstance];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.provider.allPhotos.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:
(NSIndexPath *)indexPath {
    NSString *reuseIdentifier = @"Cell";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:reuseIdentifier];
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:reuseIdentifier];
    }
    cell.textLabel.text = self.provider.allPhotos[indexPath.row];
    cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
    
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    NSString *imageName = self.provider.allPhotos[indexPath.row];
    [self performSegueWithIdentifier:@"Show" sender:imageName];
}

#pragma mark - Public

- (void)showPhotoForSearchTerm:(NSString*)searchTerm {
    if ([[PhotosProvider sharedInstance] containsPhotoForSearchTerm:searchTerm]) {
        [self performSegueWithIdentifier:@"Show" sender:searchTerm];
    }
}

#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([segue.destinationViewController isKindOfClass:[DetailViewController class]] && [sender isKindOfClass:[NSString class]]) {
        DetailViewController *detailViewController = (DetailViewController*)segue.destinationViewController;
        NSString *imageName = (NSString*)sender;
        detailViewController.image = [self.provider photoForSearchTerm:imageName];
    }
}


@end

File: SiriKit/SiriKitSwiftSample/SiriKitSwiftSample/ViewController.swift
//
//  ViewController.swift
//  SiriKitSwiftSample
//
//  Created by Nikolay Andonov on 12/7/16.
//  Copyright © 2016 Mentormate. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {

    let provider = PhotosProvider.sharedInstance
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    //MARK: - UITableViewDataSource
    
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return provider.allPhotos().count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let reuseIdentifier = "Cell"
        var cell = tableView.dequeueReusableCell(withIdentifier: reuseIdentifier)
        if cell == nil {
            cell = UITableViewCell.init(style: .default, reuseIdentifier: reuseIdentifier)
        }
        cell?.textLabel?.text = (provider.allPhotos())[indexPath.row]
        cell?.accessoryType = .disclosureIndicator
        
        return cell!
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        let imageName = (provider.allPhotos())[indexPath.row]
        performSegue(withIdentifier: "Show", sender: imageName)
        
    }
    
    //MARK: - Public
    
    func showPhotoForSearchTerm(term: String) {
        if provider.containsPhotoForSearchTerm(term: term) {
            performSegue(withIdentifier: "Show", sender: term)
        }
    }
    
    //MARK: - Navigation
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if segue.destination is DetailViewController {
            let detailViewController = segue.destination as! DetailViewController
            if let imageName = sender as? String {
                detailViewController.image = provider.photoForSearchTerm(term: imageName)
            }
        }
    }


}

